spec/defines/kmod_load_spec.rb

```diff
@@ -9,7 +9,22 @@
         facts.merge( { :augeasversion => '1.2.0' } )
       end
 
-      context 'with ensure set to present' do
+      context 'with ensure set to present on systemd' do
+        let(:params) do { :ensure => 'present', :file => '/foo/bar' } end
+        it { should contain_kmod__load('foo') }
+        it { should contain_exec('modprobe foo').with({'unless' => "egrep -q '^foo ' /proc/modules"}) }
+        let(:facts) do
+          facts.merge( { :service_provider => 'systemd' } )
+        end
+
+        it { should contain_file('/etc/modules-load.d/foo.conf').with({
+          'ensure' => 'present',
+          'mode' => '0644',
+          'content' => "# This file is managed by the puppet kmod module.\nfoo\n"
+        }) }
+      end
+
+      context 'with ensure set to present, non-systemd' do
```

Comment:
You could actually make sub-contexts inside the `ensure => present` context:

```ruby
context 'with ensure set to present' do
  context 'when on systemd' do
  end

  context 'when not on systemd' do
  end
end
```

Reply:
Ah ok, I considered that but wasn't sure if that would work, so I didn't test it. I'll refactor.